iree/turbine/kernel/wave/expansion/expansion.py

```diff
@@ -663,17 +654,15 @@ def expand_graph(
 
     add_get_results(trace)
 
-    all_nodes_reversed = list(reversed(trace.get_root_graph().nodes))
-    leaf_node_types = get_leaf_node_types(trace, all_nodes_reversed)
-    leaf_nodes = []
+    leaf_ops = [get_custom(node) for node in reversed(trace.walk(is_leaf_node))]
+    if not leaf_ops:
+        final_op = get_custom(next(reversed(trace.get_root_graph().nodes)))
```

Comment:
Just wondering could you get the last op by doing ```trace.get_root_graph()._root.prev``` ?

Reply:
Done. I'm not thrilled about using the private member here, but what I was doing wasn't working anyways (added a test for that also)